### ISSUE TITLE
Remove option breaks main slugify functionality

### ISSUE THREAD

**Issue by khromov** [simov/slugify#105]
Reproduction script:

```js
const slugify = require('slugify');

const fileName = decodeURIComponent('a%CC%8Aa%CC%88o%CC%88-123'); // åäö-123
const fileName2 = decodeURIComponent('%0A%C3%A5%C3%A4%C3%B6-123'); // åäö-123

const withoutRemove = slugify(fileName);

const withRemove = slugify(fileName, {
  remove: /[*+~.()'"!:@]/g,
});

const withRemoveAlternativeEncoding = slugify(fileName2, {
  remove: /[*+~.()'"!:@]/g,
});

console.log(withoutRemove);
console.log(withRemove);
console.log(withRemoveAlternativeEncoding);
```

Result

```
aao-123
åäö-123 # Broken
aao-123
```

I can't reproduce this when typing the "åäö" characters on my keyboard, however if I paste them from my console or a browser URL then this behaviour can be reproduced. I suspect there is a problem with different code points similar to this:
https://github.com/ipython/ipython/issues/522

**Comment by khromov** [comment]
I've updated the reproduction script to show the alternative encoding that causes the issue.

**Comment by Trott** [comment]
I haven't looked at how the implementations have diverged, but the `slug` module does not exhibit this behavior. I'm guessing/assuming it once did so I might bisect to find the change that fixed it.

**Comment by Trott** [comment]
> I haven't looked at how the implementations have diverged, but the `slug` module does not exhibit this behavior. I'm guessing/assuming it once did so I might bisect to find the change that fixed it.

Looks like this behavior has been there since the options object was introduced in slugify in 325bcaae2ad64bcf2fc13773db98b9b2283964ff. Turns out I have nothing helpful to add, so I'll delete this and my other comment to keep things tidy.

**Comment by Trott** [comment]
I'm able to reproduce this all the way back to the introduction of the `options` object in 325bcaa so this isn't anything that was once working differently and then changed. Just in case that tidbit is helpful at all.

**Comment by simov** [comment]
Yes, it is helpful, no need to delete comments. Thanks.

**Comment by simov** [comment]
With encoding one you get something like this:

```
| a  | ̊  | a  | ̈  | o  | ̈  | -  | 1  | 2  | 3  |
```

And with encoding two you get the expected characters:

```
|  |  |  å |  ä |  ö |  - |  1 |  2 |  3 |
```

Note that `å`,  `ä`,  `ö` are part of the built-in charmap and so they get replaced correctly.

When using encoding one with the default remove regex you are left with the impression that the module works correctly, but that's only because the default regex removes anything outside of `\w` and so in this case you are left with the correct characters by chance.

When using encoding one slugify cannot replace the characters with the ones in the charmap, and when you add your custom regex on top that does not filter anything outside of `\w` you endup with nothing being replaced.

**Comment by khromov** [comment]
👋 @simov 

I understand the problem. It appears this is called a "Decomposed character sequence" and is an alternate way of writing certain characters.

There is a good visual example here:
https://superpixel.ch/bugreport/safari-diacritic/

I saw [some discussion](https://stackoverflow.com/questions/11176603/how-to-avoid-browsers-unicode-normalization-when-submitting-a-form-with-unicode) regarding this on SO, pointing out Chrome / Safari as the culprit.

It feels like the current solution will work just fine in a majority of cases, since it separates the main character (which won't be filtered) from all the diacritics and such (that will be filtered). But it would be nice if it could somehow work with `remove`.

I noticed there's a [String.normalize](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/normalize) function that seems to do what we want. Maybe one option could be to apply it before any other code, but it would be hard to assess the implication of it on existing codebases.

#### Example:

Before normalization (`å` character only):

```js
console.log(Array.from(decodeURIComponent('a%CC%8A'))
  .map((v) => v.codePointAt(0).toString(16))
  .map((hex) => "\\u" + "0000".substring(0, 4 - hex.length) + hex));

> Array [ "\\u0061", "\\u030a" ] // a,  ̊
```

After normalization:

```js
console.log(Array.from(decodeURIComponent('a%CC%8A').normalize())
  .map((v) => v.codePointAt(0).toString(16))
  .map((hex) => "\\u" + "0000".substring(0, 4 - hex.length) + hex));

> Array [ "\\u00e5" ] // å - great success!
```

**Comment by simov** [comment]
Sounds good to me. Seems to be pretty safe to use since it was introduced around 6 years ago.

**Comment by simov** [comment]
Published in v1.5.0 https://github.com/simov/slugify/commit/2df63f965aaf63844a51d3c3fab7704d83a31877
